packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -164,50 +179,59 @@ export class SOQLEditorInstance {
         this.updateTextDocument(this.document, soql);
         break;
       }
+      case MessageType.UI_TELEMETRY: {
+        const { unsupported } = e.payload as TelemetryModelJson;
+        const hasUnsupported = Array.isArray(unsupported)
+          ? unsupported.length
+          : unsupported;
+        if (hasUnsupported) {
+          trackErrorWithTelemetry(
+            'syntax_unsupported',
+            JSON.stringify(e.payload)
+          ).catch(console.error);
+          const message = nls.localize('info_syntax_unsupported');
+          channelService.appendLine(message);
+        }
+        break;
+      }
       case MessageType.SOBJECT_METADATA_REQUEST: {
-        this.retrieveSObject(e.payload as string).catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object metadata for the ${e.payload} object.`
-          );
-        });
+        this.retrieveSObject(e.payload as string);
         break;
       }
       case MessageType.SOBJECTS_REQUEST: {
-        this.retrieveSObjects().catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object names.`
-          );
-        });
+        this.retrieveSObjects();
         break;
       }
       case MessageType.RUN_SOQL_QUERY: {
-        this.handleRunQuery().catch(() => {
-          channelService.appendLine(
-            `An error occurred while running the SOQL query.`
-          );
-        });
+        this.handleRunQuery();
         break;
       }
       default: {
-        console.log('message type is not supported');
+        const message = nls.localize('error_unknown_error', e.type);
+        channelService.appendLine(message);
+        trackErrorWithTelemetry('message_unknown', e.type).catch(console.error);
       }
     }
   }
 
   protected handleRunQuery(): Promise<void> {
     // Check to see if a default org is set.
     if (!workspaceContext.username) {
-      // i18n
-      const message = `No default org found. Set a default org to use SOQL Builder. Run "SFDX: Create a Default Scratch Org" or "SFDX: Authorize an Org" to set one.`;
+      const message = nls.localize('info_no_default_org');
       channelService.appendLine(message);
       vscode.window.showInformationMessage(message);
       return Promise.resolve();
     }
 
     const queryText = this.document.getText();
     return withSFConnection(async conn => {
-      const queryData = await new QueryRunner(conn).runQuery(queryText);
-      this.openQueryDataView(queryData);
+      try {
+        const queryData = await new QueryRunner(conn).runQuery(queryText);
+        this.openQueryDataView(queryData);
+      } catch (err) {
+        const message = nls.localize('error_run_soql_query', err.toString());
+        vscode.window.showErrorMessage(message);
```

Comment:
The information returned from the soql query runner api is very useful in debugging your soql statement issue.  @ekapner - once i found the useful information from the api, i gutted most of the message you carefully created.  All that is left is `We can't run the soql query."  Let me know if that's approprate.

<img width="471" alt="run-query-error" src="https://user-images.githubusercontent.com/599418/101826051-b696e500-3aeb-11eb-8371-c52e34fb482a.png">

Reply:
I'm not in love with that message but it does provide more pertinent details.